packages/app/src/files-manager/actions/upload-file.action.test.ts

```diff
@@ -0,0 +1,174 @@
+  describe("when `uploadFileRequest` request return Result.success with uploaded files list", () => {
+    beforeAll(() => {
+      ;(getPathsRequest as jest.Mock).mockResolvedValue(successGetPathResponse)
+      ;(uploadFilesRequest as jest.Mock).mockReturnValue(successUploadResponse)
+    })
+
+    afterEach(() => {
+      jest.resetAllMocks()
+    })
+
+    test("dispatch `setUploadingState` with `State.Loaded` and `getFiles` with provided directory", async () => {
+      const mockStore = createMockStore([thunk])({
+        device: {
+          deviceType: DeviceType.MuditaPure,
+        },
+      })
+
+      const {
+        meta: { requestId },
+        // AUTO DISABLED - fix me if you like :)
+        // eslint-disable-next-line @typescript-eslint/await-thenable
+      } = await mockStore.dispatch(uploadFile() as unknown as AnyAction)
+
+      expect(mockStore.getActions()).toEqual([
+        uploadFile.pending(requestId),
+        setUploadingState(State.Loading),
+        setUploadingState(State.Loaded),
+        {
+          type: pendingAction("FILES_MANAGER_GET_FILES"),
+          payload: undefined,
+        },
+        uploadFile.fulfilled(undefined, requestId),
+      ])
+
+      expect(getPathsRequest).toHaveBeenLastCalledWith(
+        getFilesPathsResponseMock
+      )
+      expect(uploadFilesRequest).toHaveBeenLastCalledWith({
+        directory: DeviceDirectory.Music,
+        paths: pathsMock,
+      })
+    })
+  })
+
+  describe("when `uploadFileRequest` request return Result.failed", () => {
+    beforeAll(() => {
+      ;(getPathsRequest as jest.Mock).mockResolvedValue(successGetPathResponse)
+      ;(uploadFilesRequest as jest.Mock).mockReturnValue(failedUploadResponse)
+    })
+
+    afterEach(() => {
+      jest.resetAllMocks()
+    })
+
+    test("dispatch `setUploadingState` with `State.Loaded` and `getFiles` with provided directory", async () => {
+      const mockStore = createMockStore([thunk])({
+        device: {
+          deviceType: DeviceType.MuditaPure,
+        },
+      })
+
+      const {
+        meta: { requestId },
+        // AUTO DISABLED - fix me if you like :)
+        // eslint-disable-next-line @typescript-eslint/await-thenable
+      } = await mockStore.dispatch(uploadFile() as unknown as AnyAction)
+
+      expect(mockStore.getActions()).toEqual([
+        uploadFile.pending(requestId),
+        setUploadingState(State.Loading),
+
+        uploadFile.rejected(testError, requestId, undefined, { ...errorMock }),
+      ])
+
+      expect(getPathsRequest).toHaveBeenLastCalledWith(
+        getFilesPathsResponseMock
+      )
+      expect(uploadFilesRequest).toHaveBeenLastCalledWith({
+        directory: DeviceDirectory.Music,
+        paths: pathsMock,
+      })
+    })
+  })
+})
+
+describe("when `getPathRequest` request return Result.failed", () => {
+  describe("when `uploadFileRequest` request return Result.success with uploaded files list", () => {
```

Comment:
Seems to me that this describe is unnecessary - when `getPathRequest` returns failure, `uploadFileRequest` is not called at all.

Reply:
https://github.com/mudita/mudita-center/pull/914/commits/2d56a5d32198dd77433655a6048054821f713766